cmd/cayley/command/repl.go

```diff
@@ -101,9 +102,17 @@ func NewQueryCmd() *cobra.Command {
 		Aliases: []string{"qu"},
 		Short:   "Run a query in a specified database and print results.",
 		RunE: func(cmd *cobra.Command, args []string) error {
+			var querystr string
 			if len(args) != 1 {
-				return fmt.Errorf("expected query as an argument")
+				bytes, err := ioutil.ReadAll(os.Stdin)
+				if err != nil {
+					panic(err)
+				}
+				querystr = string(bytes)
+			} else {
+				querystr = args[0]
 			}
+			fmt.Printf("Query to run is :%s\n", querystr)
```

Comment:
This should be `log.Printf` (uses stderr), since you are mixing your message with an output of a query itself (both are stdout).

Also change a message to something simple, like `Query:` or `Running:`. Also it's a good idea to place line break after `:`.